src/components/Editor/index.tsx

```diff
@@ -138,16 +135,22 @@ const Editor = () => {
     const {mode, systemMode} = useColorScheme();
 
     const beginLineNumToLogEventNum = useViewStore((state) => state.beginLineNumToLogEventNum);
+    const isPrettified = useViewStore((state) => state.isPrettified);
+    const updateIsPrettified = useViewStore((state) => state.updateIsPrettified);
+
     const logData = useViewStore((state) => state.logData);
+
+    const logEventNum = useViewStore((state) => state.logEventNum);
+    const setLogEventNum = useViewStore((state) => state.setLogEventNum);
+
     const loadPageByAction = useViewStore((state) => state.loadPageByAction);
-    const {isPrettified, logEventNum} = useContext(UrlContext);
 
     const [lineNum, setLineNum] = useState<number>(1);
     const beginLineNumToLogEventNumRef = useRef<BeginLineNumToLogEventNumMap>(
         beginLineNumToLogEventNum
     );
     const editorRef = useRef<Nullable<monaco.editor.IStandaloneCodeEditor>>(null);
-    const isPrettifiedRef = useRef<boolean>(isPrettified ?? false);
+    const isPrettifiedRef = useRef<boolean>(isPrettified);
```

Comment:
similar to https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , we may not no long need to copy states into refs. instead, we can directly read them from the Zustand stores.